src/Types/UnbalancedTransaction.purs

```diff
@@ -121,8 +126,9 @@ instance DecodeJson PubKeyHash where
 payPubKeyVkey :: PaymentPubKey -> Vkey
 payPubKeyVkey (PaymentPubKey pk) = Vkey pk
 
-payPubKeyRequiredSigner :: PaymentPubKey -> RequiredSigner
-payPubKeyRequiredSigner pk = RequiredSigner $ payPubKeyVkey pk
+payPubKeyRequiredSigner :: PaymentPubKey -> Maybe RequiredSigner
+payPubKeyRequiredSigner (PaymentPubKey (PublicKey bech32)) =
+  RequiredSigner <<< publicKeyHash <$> publicKeyFromBech32 bech32
```

Comment:
> Shouldn't bech32 wrapped in PublicKey be guaranteed to be a valid one? Can't we make it so?

No guarantee at the moment.
Are you concerned with one more variant of `MkUnbalancedTxError` I added? If not, I will make a separate PR for that, because there are many other types with no validity guarantees.

Technically, we have no reason to use Bech32String type at all if we want smart constructors, because we will have to call CSL for validation. We could just use the foreign type, similarly to how it was done with `Address`.

I think I'll do exactly that if there are no objections.